runtime/mux.go

```diff
@@ -75,6 +76,30 @@ type ServeMux struct {
 // ServeMuxOption is an option that can be given to a ServeMux on construction.
 type ServeMuxOption func(*ServeMux)
 
+// ForwardResponseRewriter is the signature of a function that is capable of rewriting messages
+// before the are forwarded in unary response, stream response, and error responses.
+type ForwardResponseRewriter func(ctx context.Context, response proto.Message) (interface{}, error)
+
+// WithForwardResponseRewriter returns a ServeMuxOption that allows for implementers to insert logic
+// that can rewrite the final response before it is forwarded.
+//
+// The response rewriter function is called during unary message forwarding, stream message
+// forwarding and when errors are being forwarded.
+func WithForwardResponseRewriter(fwdResponseRewriter ForwardResponseRewriter) ServeMuxOption {
+	return func(sm *ServeMux) {
+		sm.forwardResponseRewriter = func(ctx context.Context, response proto.Message) (interface{}, error) {
+			result, err := fwdResponseRewriter(ctx, response)
+			if err != nil {
+				return nil, err
+			}
+			if rb, ok := result.(responseBody); ok {
+				return rb.XXX_ResponseBody(), nil
+			}
```

Comment:
It's a bit confusing to me to have business logic like this in one of these options. Is there any reason we can't return this to the respective handler functions instead?

Reply:
Sure. That sounds better.